src/schematools/datasetcollection.py

```diff
@@ -49,8 +63,29 @@ def get_dataset(self, dataset_id: str) -> DatasetSchema:
         try:
             return self.datasets_cache[dataset_id]
         except KeyError:
-            dataset = self._load_dataset(dataset_id)
+            dataset = self._load_dataset(dataset_id, prefetch_related=prefetch_related)
             if dataset is None:
                 raise ValueError(f"Dataset {dataset_id} is missing.") from None
             self.add_dataset(dataset)
             return dataset
+
+
+def set_schema_loader(schema_url: Union[URL, str]) -> None:
+    """Initialize the schema loader at module load time.
+
+    schema_url:
+        Location where the schemas can be found. This
+        can be a web url, or a filesystem path.
+    """
+    dataset_collection = DatasetCollection()
+    loader: Optional[loaders.SchemaLoader] = None  # pleasing mypy
+    if urlparse(schema_url).scheme in ("http", "https"):
+        loader = loaders.URLSchemaLoader(schema_url)
+    else:
+        loader = loaders.FileSystemSchemaLoader(schema_url)
+    dataset_collection.set_schema_loader(loader)
+
+
+# The scheme loader is initialized from the `SCHEMA_URL` environment variable,
+# or from the DEFAULT_SCHEMA_URL.
+set_schema_loader(os.environ.get("SCHEMA_URL", DEFAULT_SCHEMA_URL))
```

Comment:
If this is meant to run only at module init time, shouldn't it be more private?

```suggestion
def _set_schema_loader(schema_url: Union[URL, str]) -> None:
    """Initialize the schema loader at module load time.

    schema_url:
        Location where the schemas can be found. This
        can be a web url, or a filesystem path.
    """
    dataset_collection = DatasetCollection()
    loader: Optional[loaders.SchemaLoader] = None  # pleasing mypy
    if urlparse(schema_url).scheme in ("http", "https"):
        loader = loaders.URLSchemaLoader(schema_url)
    else:
        loader = loaders.FileSystemSchemaLoader(schema_url)
    dataset_collection.set_schema_loader(loader)


# The scheme loader is initialized from the `SCHEMA_URL` environment variable,
# or from the DEFAULT_SCHEMA_URL.
_set_schema_loader(os.environ.get("SCHEMA_URL", DEFAULT_SCHEMA_URL))
```

Reply:
Good point to express that is it is not meant to be called externally by flagging it private with `_`.

Reply:
# This call is done at module load time, to have an initial value for the schemaloader.
# If needed, an alternative schemaloader can be injected into `DatasetCollection` at runtime.